Mathlib/Condensed/Discrete/LocallyConstant.lean

```diff
@@ -0,0 +1,429 @@
+  apply injective_of_mono (sigmaComparison X (fun a ↦ (fiber f a).1))
+  ext a
+  specialize h a
+  rw [← sigmaComparison_comp_sigmaIso] at h
+  exact h
+
+lemma incl_of_counitAppApp [PreservesFiniteProducts Y] [HasExplicitFiniteCoproducts.{u} P]
+    (a : Fiber f) : Y.map (sigmaIncl f a).op (counitAppApp S Y f) = counitAppAppImage f a := by
+  rw [← sigmaComparison_comp_sigmaIso, Functor.mapIso_hom, Iso.op_hom, types_comp_apply]
+  simp only [counitAppApp, Functor.mapIso_inv, ← Iso.op_hom, types_comp_apply,
+    ← FunctorToTypes.map_comp_apply, Iso.inv_hom_id, FunctorToTypes.map_id_apply]
+  exact congrFun (inv_hom_id_apply (asIso (sigmaComparison Y (fun a ↦ (fiber f a).1)))
+    (counitAppAppImage f)) _
+
+variable {T : CompHausLike.{u} P} (g : T ⟶ S)
+
+/--
+This is an auxiliary definition, the details do not matter. What's important is that this map exists
+so that the lemma `incl_comap` works.
+-/
+def componentHom (a : Fiber (f.comap g)) :
+    fiber _ a ⟶ fiber _ (Fiber.mk f (g a.preimage)) where
+  toFun x := ⟨g x.val, by
+    simp only [Fiber.mk, Set.mem_preimage, Set.mem_singleton_iff]
+    convert map_eq_image _ _ x
+    exact map_preimage_eq_image_map _ _ a⟩
+  continuous_toFun := by
+    exact Continuous.subtype_mk (Continuous.comp g.continuous continuous_subtype_val) _
+    -- term mode gives "unknown free variable" error.
+
+lemma incl_comap {S T : (CompHausLike P)ᵒᵖ}
+    (f : LocallyConstant S.unop (Y.obj (op (CompHausLike.of P PUnit.{u+1}))))
+      (g : S ⟶ T) (a : Fiber (f.comap g.unop)) :
+        g ≫ (sigmaIncl (f.comap g.unop) a).op =
+          (sigmaIncl f _).op ≫ (componentHom f g.unop a).op :=
+  rfl
+
+/-- The counit is natural in `S : CompHausLike P` -/
+@[simps!]
+noncomputable def counitApp [HasExplicitFiniteCoproducts.{u} P]
+    (Y : (CompHausLike.{u} P)ᵒᵖ ⥤ Type max u w) [PreservesFiniteProducts Y] :
+    (functorToPresheaves.obj (Y.obj (op (CompHausLike.of P PUnit.{u+1})))) ⟶ Y where
+  app := fun ⟨S⟩ ↦ counitAppApp S Y
+  naturality := by
+    intro S T g
+    ext f
+    apply presheaf_ext (f.comap g.unop)
+    intro a
+    simp only [op_unop, functorToPresheaves_obj_obj, types_comp_apply, functorToPresheaves_obj_map,
+      incl_of_counitAppApp, ← FunctorToTypes.map_comp_apply, incl_comap]
+    simp only [FunctorToTypes.map_comp_apply, incl_of_counitAppApp]
+    simp only [counitAppAppImage, ← FunctorToTypes.map_comp_apply, ← op_comp,
+      terminal.comp_from]
+    apply congrArg
+    exact image_eq_image_mk (g := g.unop) (a := a)
+
+variable (P) (X : TopCat.{max u w})
+    [HasExplicitFiniteCoproducts.{0} P] [HasExplicitPullbacks P]
+    (hs : ∀ ⦃X Y : CompHausLike P⦄ (f : X ⟶ Y), EffectiveEpi f → Function.Surjective f)
+
+/-- `locallyConstantIsoContinuousMap` is a natural isomorphism. -/
+noncomputable def functorToPresheavesIsoTopCatToSheafCompHausLike (X : Type (max u w)) :
+    functorToPresheaves.{u, w}.obj X ≅
+      ((topCatToSheafCompHausLike P hs).obj (TopCat.discrete.obj X)).val :=
```

Comment:
I'm going to merge this as is. But I think a follow-up PR could rename `topCatToSheafCompHausLike` to `TopCat.toSheafCompHausLike`. That would allow you to make several names a bit shorter.

Reply:
The name `TopCat.toSheafCompHausLike` already exists as the object part of that functor (which is more useful because of dot notation). But there are indeed some places in this file where that can be used, I opened #17075